test/e2e/reporter.go

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/openshift-kni/k8sreporter"
+	"github.com/pkg/errors"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	metallbv1beta2 "go.universe.tf/metallb/api/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const frrContainerName = "frr"
+
+func InitReporter(kubeconfig, path, namespace string) *k8sreporter.KubernetesReporter {
+	// When using custom crds, we need to add them to the scheme
+	addToScheme := func(s *runtime.Scheme) error {
+		err := metallbv1beta1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = metallbv1beta2.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// The namespaces we want to dump resources for (including pods and pod logs)
+	dumpNamespace := func(ns string) bool {
+		return ns == namespace
+	}
+
+	// The list of CRDs we want to dump
+	crds := []k8sreporter.CRData{
```

Comment:
I guess you can add also the ovnk crds / namespaces, it won't hurt

Reply:
now adding ovnk crds for the collection, but just realized this would collect only from metallb-system namespace which doesn't make sense as it won't have any of ovnk cr objects. 
will improve method so that it would collect logs from more relevant namespaces like test ns, ovn-kubernetes.

Reply:
For that there's the dumpNamespace above

Reply:
yes, now dumping logs for relevant namespaces.